TheaterSchedule.BLL/Services/ImageService.cs

```diff
@@ -1,16 +1,188 @@
+                    {
+                        Image = ImageToBase64(image),
+                        ImageFormat = GeMimeTypeFromImageByteArray(image)
+                    };
+                });
+            }
+
+            return imageDTO;
+        }
+
+        public async Task<ImageBytesDTO> LoadPerformanceMainImageBytesAsync(int id)
+        {
+            ImageBytesDTO imageDTO = null;
+            byte[] image = await TryAddPerformanceImageToCacheAsync(id);
+
+            if (image != null)
+            {
+                await Task.Run(() =>
+                {
+                    imageDTO = new ImageBytesDTO()
+                    {
+                        Image = image,
+                        ImageFormat = GeMimeTypeFromImageByteArray(image)
+                    };
+                });
+            }
+
+            return imageDTO;
+        }
+
+        public async Task<List<ImageBase64DTO>> LoadPerformanceGalleryBase64Async(int id)
+        {
+            List<ImageBase64DTO> imagesDTO = new List<ImageBase64DTO>();
+            IEnumerable<byte[]> images = await TryAddGalleryImagesToCacheAsync(id);
+
+            await Task.Run(() =>
+            {
+                foreach (var image in images)
+                {
+                    imagesDTO.Add(new ImageBase64DTO()
+                    {
+                        Image = ImageToBase64(image),
+                        ImageFormat = GeMimeTypeFromImageByteArray(image)
+                    });
+                }
+            });
+
+            return imagesDTO;
+        }
+
+        public async Task<List<ImageBytesDTO>> LoadPerformanceGalleryBytesAsync(int id)
+        {
+            List<ImageBytesDTO> imagesDTO = new List<ImageBytesDTO>();
+            IEnumerable<byte[]> images = await TryAddGalleryImagesToCacheAsync(id);
+
+            await Task.Run(() =>
+            {
+                foreach (var image in images)
+                {
+                    imagesDTO.Add(new ImageBytesDTO()
+                    {
+                        Image = image,
+                        ImageFormat = GeMimeTypeFromImageByteArray(image)
+                    });
+                }
+            });
+
+            return imagesDTO;
+        }
+
+
+        public async Task<IEnumerable<byte[]>> TryAddGalleryImagesToCacheAsync(int performanceId)
+        {
+            bool needUpdateCache = false;
+            IEnumerable<byte[]> images = new List<byte[]>();
+            int imgsNumber = await imageRepository.GetGalleryImagesCountAsync(performanceId);
+
+            for (int nImg = 0; nImg < imgsNumber; nImg++)
+            {
+                byte[] image = null;
+                string key = typeof(Performance).Name + performanceId.ToString() + typeof(GalleryImage).Name + nImg;
+
+                if (!cache.TryGetValue(key, out image))
+                {
+                    images = await imageRepository.GetGalleryImagesAsync(performanceId);
+                    needUpdateCache = true;

```

Comment:
put update logic here and remove bool flag
For better readability you also can extract code to method like 'SetExpirationTime'